### PR TITLE
Adjust 'test_snmp_phy_entity' test case to support any number of lanes

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
### Description of PR

Summary:
Adjust test case to any type of lanes per cable and not fixed hard coded value of 4 lanes due to recent support of QSFP-DD. 

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?

Adjust the test case to support cables which has any number of lanes.

#### How did you do it?

Read lane data from redis and compare it with snmp output.

#### How did you verify/test it?

Manually run the test case

#### Any platform specific information?

N/A

#### Supported testbed topology if it's a new test case?

N/A

### CODE REVIEW

**Comment by keboliu** [comment]
retest vsimage please

**Comment by liat-grozovik** [comment]
@jleveque  and @qiluo-msft  could you please review this fix/enhancement?

**Review by jleveque — APPROVED** [review]

File: tests/snmp/test_snmp_phy_entity.py [hunk 1]
```diff
@@ -74,20 +75,12 @@
...
-                        SENSOR_TYPE_PORT_RX_POWER + 3,
-                        SENSOR_TYPE_PORT_RX_POWER + 4,
-                        SENSOR_TYPE_PORT_TX_BIAS + 1,
-                        SENSOR_TYPE_PORT_TX_BIAS + 2,
-                        SENSOR_TYPE_PORT_TX_BIAS + 3,
-                        SENSOR_TYPE_PORT_TX_BIAS + 4,
-                        SENSOR_TYPE_VOLTAGE]
+XCVR_SENSOR_PATTERN = {
+    'temperature': {'sort_factor': 0, 'oid_base': SENSOR_TYPE_TEMP, 'extract_line_number': False},
+    'voltage': {'sort_factor': 9000, 'oid_base': SENSOR_TYPE_VOLTAGE, 'extract_line_number': False},
```
Comment (qiluo-msft) [inline tests/snmp/test_snmp_phy_entity.py:80]: >9000 [](start = 31, length = 4)

How do you choose this number? If it is based on other code, could you provide some code comment to refer that code?

File: tests/snmp/test_snmp_phy_entity.py [hunk 2]
```diff
@@ -535,40 +561,13 @@ def redis_hgetall(duthost, db_id, key):
...
-    field_names = content.split('\n')
-    for field_name in field_names:
-        field_name = field_name.strip()
-        cmd = 'sonic-db-cli {} HGET \"{}\" \"{}\"'.format(db_id, key, field_name)
-        output = duthost.shell(cmd)['stdout'].strip()
-        result[field_name] = output
+        return {}
 
-    return result
+    return eval(content)
```
Comment (qiluo-msft) [inline tests/snmp/test_snmp_phy_entity.py:570]: >eval [](start = 11, length = 4)

It is dangerous to `eval`. I found a possible solution at [https://stackoverflow.com/a/13165500/2514803](https://stackoverflow.com/a/13165500/2514803)

**Review by qiluo-msft — CHANGES_REQUESTED** [review]
As comment

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**